lib/nvme/nvme.c

```diff
@@ -248,6 +248,9 @@ nvme_probe_one(enum spdk_nvme_transport transport, spdk_nvme_probe_cb probe_cb,
 		ctrlr->probe_info = *probe_info;
 
 		TAILQ_INSERT_TAIL(&g_spdk_nvme_driver->init_ctrlrs, ctrlr, tailq);
+	} else {
+		/* The device is not supported */
+		return 1;
```

Comment:
I merged this part of the fix (you are right, it is still in nvme.c, my mistake) as ea3a2772ac78f3646d0c98e9c7995bcf5ed1ee81 - thanks.